File: backendmealplan/src/main/java/org/backendmealplan/backendmealplan/beans/UserFeedback.java
package org.backendmealplan.backendmealplan.beans;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;


@NoArgsConstructor
@AllArgsConstructor
@Entity
@Data
@Table(name = "feedbacks")
public class UserFeedback {



    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long feedbackId;
    @ManyToOne
    @JoinColumn(name = "meal_id")
    private Meal meal;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private Boolean isOnIt;
    private Date date;
    private int rating=-1;
    private String feedbackText;

}

File: frontendmealplan/src/app/mealplan/models/UserFeedback.ts
import { Meal } from "./Meal";
import { User } from "./User";

export class UserFeedback {
  constructor(    public feedbackId?: number,
                  public meal?: Meal,
                  public user?:User,
                  public isOnIt?: boolean,
                  public date?: Date,
                  public rating?: number,
                  public feedbackText?: string) {
  }
}